crates/rolldown_plugin/src/plugin_driver/hook_orders.rs

```diff
@@ -0,0 +1,119 @@
+        p.call_resolve_id_meta()
+      }),
+      order_by_resolve_dynamic_import_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_resolve_dynamic_import_meta()
+      }),
+      order_by_load_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| p.call_load_meta()),
+      order_by_transform_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_transform_meta()
+      }),
+      order_by_module_parsed_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_module_parsed_meta()
+      }),
+      order_by_build_end_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_build_end_meta()
+      }),
+      order_by_render_start_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_render_start_meta()
+      }),
+      order_by_banner_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_banner_meta()
+      }),
+      order_by_footer_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_footer_meta()
+      }),
+      order_by_intro_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| p.call_intro_meta()),
+      order_by_outro_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| p.call_outro_meta()),
+      order_by_render_chunk_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_render_chunk_meta()
+      }),
+      order_by_augment_chunk_hash_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_augment_chunk_hash_meta()
+      }),
+      order_by_render_error_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_render_error_meta()
+      }),
+      order_by_generate_bundle_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_generate_bundle_meta()
+      }),
+      order_by_write_bundle_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_write_bundle_meta()
+      }),
+      order_by_close_bundle_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_close_bundle_meta()
+      }),
+      order_by_watch_change_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_watch_change_meta()
+      }),
+      order_by_close_watcher_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_close_watcher_meta()
+      }),
+      order_by_transform_ast_meta: Self::sort_plugins_by_hook_meta(index_plugins, |p| {
+        p.call_transform_ast_meta()
+      }),
+    }
+  }
+
+  fn sort_plugins_by_hook_meta(
+    index_plugins: &IndexPluginable,
+    get_hook_meta: impl Fn(&SharedPluginable) -> Option<PluginHookMeta>,
+  ) -> Vec<PluginIdx> {
+    let mut pre_plugins = vec![];
+    let mut normal_plugins = vec![];
+    let mut post_plugins = vec![];
+
+    for (idx, plugin) in index_plugins.iter_enumerated() {
+      let meta = get_hook_meta(plugin);
+      if let Some(meta) = meta {
```

Comment:
Not relevant to this pr, but: 
It seems three arrays are not needed, maybe a simple stable sort should enough?